Us598 and US597: Combo3 level finalization (plz build after merge)

jmolive8 opened the pull request:
Completed Combo3 level. Finished return path after end cave and reworked segments where you drop an iron rectangle into a spike hole to a new puzzle where you use an iron square to open a door. Could we have a new build after this is merged so playtesters have access to the new features?

Added ceiling tiles to Landing to prevent the player from flying off the edge of the map with the upgraded thruster and  adjusted the camera boundary so the ceiling tiles are out of reach of the camera and it doesn't look like you are underground. Also adjusted camera boundaries on EMagnet, and Combo3.

Removed needless scripts, prefabs and temp objects on scenes. Renamed a few prefabs and reorganized folders slightly.

EDIT: Made transition cave to Combo3 hidden behind some fake terrain and a locked door on Landing. Also shifted the player and ship's position so that players see the hollowed out area suggesting that it's there but will not be able to do anything with it since they don't have all the tools yet. Let me know if you guys don't like this setup.

EDIT 2: Added US597. Renamed Iron objects to Magnetized and Basic Moving Hazard to Moving Spiked Ball in all scenes. Iron Layer was renamed to Magnetized. Iron Elements were left unchanged.

- [x] The code compiles
- [x] The code has been developer-tested
- [x] The script and each function has a description
- [x] The code follows guidelines listed in Quality Control Practices

LopezRoman commented:
Play tested and works well. The only issue I saw was that now loading from Title Scene loads the player at the incorrect starting location, now loads at the cave that leads to Combo 3 scene.

mcmyers4 commented on Psyche/Assets/Prefabs/Pickupables/Tools/SolarArray.prefab:
Please refrain from using SolarPanel anymore. See US613.

jmolive8 replied:
Idk why that's saying I renamed it to SolarPanel. I made sure it was named SolarArray and I don't see anything on my end still named SolarPanel.

jmolive8 replied:
Okay so apparently I just fixed it but I don't really know what I did lol. It might have been the pickup in the Landing scene wasn't in sync with the prefab cause I did have the blue bar saying it was changed and I reverted those changes.

jmolive8 commented:
I removed my changes to SceneStateManager because trying a code workaround for spawning on the Combo3 In Cave vs the initial spawn point was getting needlessly complicated and wasn't really working. So I hid the Combo3 cave behind some fake terrain as detailed above and changed the Combo3 In Cave's layering so it doesn't conflict with the initial spawn point.

LopezRoman commented:
Works well now. From Title player lands at the proper starting location, and from Combo 3 player lands at the appropriate cave object. 

- [x] Comments and descriptions are comprehensible and add to the maintainability of the code
- [x] Comments are not too verbose or numerous
- [x] Code compiles
- [x] The goal of the US has been fulfilled
- [x] There are no merging conflicts
- [x] The code follows guidelines listed in Quality Control Practices

mcmyers4 commented:
Transitioning from combo3 back to combo2 is doing some weird stuff

mcmyers4 reviewed (changes requested):
![image](https://github.com/MissionToPsyche/tungsten_12a_web_game-se/assets/122760299/95e312ab-c04e-49f5-a3d4-eddf2c559308)

![image](https://github.com/MissionToPsyche/tungsten_12a_web_game-se/assets/122760299/50b66355-a155-49de-bd87-77c5f8f60ab9)

JoshBenn commented:
Holding off on merge until this is either addressed or viewed/handled appropriately by the owning dev

LopezRoman commented:
> Transitioning from combo3 back to combo2 is doing some weird stuff

![image](https://github.com/MissionToPsyche/tungsten_12a_web_game-se/assets/54126100/16019f65-7fdd-4afc-b024-3537feca82a9)

Proper game object is not found when transitioning back to Combo 2. The problem comes from the detecting that we're coming from Combo 3 which now looks for 'cave' tagged transition objects. Easy fix is to set the cave in Combo 2 to tag 'cave' instead of default. 

![image](https://github.com/MissionToPsyche/tungsten_12a_web_game-se/assets/54126100/eda752ec-fbff-4c1f-aef2-4a2d46d3bfce)

![image](https://github.com/MissionToPsyche/tungsten_12a_web_game-se/assets/54126100/0d75a47a-4a34-4d72-9181-d3888a5c9fbe)
